sqld/src/lib.rs

```diff
@@ -56,514 +62,96 @@ mod query;
 mod query_analysis;
 mod query_result_builder;
 mod replication;
-pub mod rpc;
 mod stats;
 #[cfg(test)]
 mod test;
 mod utils;
-pub mod version;
 
 const MAX_CONCURRENT_DBS: usize = 128;
 const DB_CREATE_TIMEOUT: Duration = Duration::from_secs(1);
 const DEFAULT_NAMESPACE_NAME: &str = "default";
 const DEFAULT_AUTO_CHECKPOINT: u32 = 1000;
 
-#[derive(clap::ValueEnum, Clone, Debug, PartialEq)]
-pub enum Backend {
-    Libsql,
-}
-
 type Result<T, E = Error> = std::result::Result<T, E>;
 
-#[derive(Debug, Clone)]
-pub struct Config {
-    pub db_path: PathBuf,
-    pub extensions_path: Option<PathBuf>,
-    pub http_addr: Option<SocketAddr>,
-    pub enable_http_console: bool,
-    pub http_auth: Option<String>,
-    pub http_self_url: Option<String>,
-    pub hrana_addr: Option<SocketAddr>,
-    pub admin_addr: Option<SocketAddr>,
-    pub auth_jwt_key: Option<String>,
-    pub backend: Backend,
-    pub writer_rpc_addr: Option<String>,
-    pub writer_rpc_tls: bool,
-    pub writer_rpc_cert: Option<PathBuf>,
-    pub writer_rpc_key: Option<PathBuf>,
-    pub writer_rpc_ca_cert: Option<PathBuf>,
-    pub rpc_server_addr: Option<SocketAddr>,
-    pub rpc_server_tls: bool,
-    pub rpc_server_cert: Option<PathBuf>,
-    pub rpc_server_key: Option<PathBuf>,
-    pub rpc_server_ca_cert: Option<PathBuf>,
-    pub bottomless_replication: Option<bottomless::replicator::Options>,
+pub struct Server<C = HttpConnector, A = AddrIncoming> {
+    pub path: Arc<Path>,
+    pub db_config: DbConfig,
+    pub user_api_config: UserApiConfig<A>,
```

Comment:
Idk, it's not only an HTTP server, if anything, it's a `SqldServer`, but since we're changing the name, I avoid including sqld into anything. Any other ideas?